### PR TITLE
Add publishing scripts

### PR DESCRIPTION
**Pull request by Jefiozie**
With these changes we can run the release-it commands via: `npm run release:lib <package>`. running this will bump the packages based on the git strategy. After running this we should manually run the `npm run build.all.librarys ` command and publish it to npm. 

For now I disabled tagging and publishing to npm. As I don't see a easy option on how we should configure this for all the different packages. Maybe  you know this @manekinekko?

### CODE REVIEW

**Comment by manekinekko** [comment]
> For now I disabled tagging and publishing to npm. As I don't see a easy option on how we should configure this for all the different packages. Maybe you know this @manekinekko?

What about if we create one github repo per library build? The purpose of this *special* repo would be to host the built source of a specific package and track the releases/tags. For instance, we could have:

- `angular-codegen-build`
- `angular-codegen-build`

File: .vscode/settings.json [hunk 1]
```diff
@@ -27,5 +27,6 @@
     "**/bazel-out": true,
     "**/dist": true
   },
-  "peacock.color": "#ff2c70"
+  "peacock.color": "#ff2c70",
+  "terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe"
```
Comment (manekinekko) [inline .vscode/settings.json]: I think this line should be removed bc it's platform-specific.

File: projects/angular-codegen/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@xlayers/angular-codegen",
-  "version": "0.0.1",
+  "version": "0.0.2-beta.0",
```
Comment (manekinekko) [inline projects/angular-codegen/package.json]: OK for now. We will discuss the correct versioning strategy later.

File: scripts/build-all-librarys.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { cmd, runTask } from './utils';
+import { cmd, runTask, packages } from './utils';
 
 export async function buildAllLibrarys() {
```
Comment (manekinekko) [inline scripts/build-all-librarys.ts]:
```suggestion
 export async function buildAllLibraries() {
```

File: scripts/publish-library.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { cmd, runTask, packages } from './utils';
+
+export async function buildAllLibrarys() {
```
Comment (manekinekko) [inline scripts/publish-library.ts]:
```suggestion
export async function buildAllLibraries() {
```

**Comment by manekinekko** [comment]
I think all occurrences of `librarys` should be renamed to `libraries`.

File: .vscode/settings.json (hunk 1 above)
Comment (Jefiozie) [reply .vscode/settings.json]: Oops 😅

**Comment by Jefiozie** [comment]
I like the idea of the special repo where we publish the source.
Great idea!

Could you make the repos for this? I believe I cannot because of rights

**Comment by Jefiozie** [comment]
So I added some stuff for the push to build repo's. I'm still facing some issues that I need to figure out but I think the idea is pretty clear.

File: scripts/utils.ts [hunk 5]
```diff
@@ -8,13 +9,13 @@ export async function runTask(name: string, taskFn: () => Promise<any>) {
     const spinner = ora(name);
 
     try {
-        spinner.start();
+        // spinner.start();
```
Comment (Jefiozie) [inline scripts/utils.ts]: Going to remove this later :)

File: scripts/build-all-libraries.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
-import { cmd, runTask } from './utils';
+import { cmd, runTask, packages } from './utils';
```
Comment (manekinekko) [inline scripts/build-all-libraries.ts]:
```suggestion
import { cmd, runTask } from './utils';
```

**Comment by manekinekko** [comment]
@Jefiozie looks interesting. I have to feedback:
- I can't remember whether we are gonna publish all libraries under the same version or separate version? - Can you add a readme about the publish process for further reference?

Cheers.

**Comment by Jefiozie** [comment]
HI @manekinekko, so the idea we had is to publish all libraries in a separate way. We can release all of the packages separately now with their own version. This makes it easy to release hotfixes for example.

>Can you add a readme about the publish process for further reference?

Sure thing! Maybe it is better to add it to the wiki, what do you think?

**Comment by Jefiozie** [comment]
Note to @xlayers/maintainers only let the script push to nightly/build repo's.

**Comment by manekinekko** [comment]
@Jefiozie is this PR ready for a review?

**Comment by Jefiozie** [comment]
@manekinekko No, I'm not yet ready for review. I'm having a hard time to find time to work on this. I hope I can find some time in the upcoming days.

**Comment by Jefiozie** [comment]
A small update, I've had a look at it again but for some reason on my new machine (windows), I have some different behavior than I had before on my osx. I need to dig a bit deeper into why this is and how we could resolve it.

**Comment by Jefiozie** [comment]
Let me update the PR and make documentation for this after that we will do a review.

**Pull request closed without merging**